pkg/workloadmeta/collectors/internal/ecs/ecs.go

```diff
@@ -63,7 +65,13 @@ func (c *collector) Start(ctx context.Context, store workloadmeta.Store) error {
 		return err
 	}
 
+	// This only exists to allow overriding for testing
+	c.metaV3or4 = func(metaURI, metaVersion string) v3or4.Client {
+		return v3or4.NewClient(metaURI, metaVersion)
+	}
```

Comment:
`v3or4.NewClient` does nothing.
Invoking it from here rather than from line 256 shouldn’t be an issue.

```suggestion
	// This only exists to allow overriding for testing
	c.metaV3or4 = v3or4.NewClient(metaURI, metaVersion)
```

Reply:
The problem with that – and the reason I made this a function returning a `v3or4.Client` – is we don't have `metaURI` and `metaVersion` in scope here. Those are determined dynamically from the container env vars, so we end up creating a new client each time we run `getResourceTags`.

We could change the methods to require `metaURI` and `metaVersion` as parameters each time, e.g.

```
- func (c *client) GetContainer(ctx context.Context) (*Container, error) {
+func (c *client) GetContainer(ctx context.Context) (metaUri, metaVersion string, *Container, error) {
```

.. but this is inconsistent with the other clients and is a little messy.